pro/src/components/IndividualOfferForm/OfferLocation/OfferLocation.tsx

```diff
@@ -0,0 +1,132 @@
+import { useField, useFormikContext } from 'formik'
+import React, { useState } from 'react'
+
+import { VenueListItemResponseModel } from 'apiClient/v1'
+import { AddressSelect } from 'components/Address/Address'
+import { AddressManual } from 'components/AddressManual/AddressManual'
+import { FormLayout } from 'components/FormLayout/FormLayout'
+import fullBackIcon from 'icons/full-back.svg'
+import fullNextIcon from 'icons/full-next.svg'
+import { Button } from 'ui-kit/Button/Button'
+import { ButtonVariant } from 'ui-kit/Button/types'
+import { RadioButton } from 'ui-kit/form/RadioButton/RadioButton'
+import { TextInput } from 'ui-kit/form/TextInput/TextInput'
+
+import { IndividualOfferFormValues } from '../types'
+
+import { OFFER_LOCATION } from './constants'
+import styles from './OfferLocation.module.scss'
+
+export interface OfferLocationProps {
+  venue: VenueListItemResponseModel | undefined
+}
+
+export const OfferLocation = ({ venue }: OfferLocationProps): JSX.Element => {
+  const formik = useFormikContext<IndividualOfferFormValues>()
+
+  const [showOtherAddress, setShowOtherAddress] = useState(
+    formik.values.offerlocation === OFFER_LOCATION.OTHER_ADDRESS
+  )
+  const [manuallySetAddress, , { setValue: setManuallySetAddress }] =
+    useField('manuallySetAddress')
+
+  const onManuallySetAddress = async () => {
+    const newValue = !manuallySetAddress.value
+    await setManuallySetAddress(newValue)
+    if (newValue) {
+      const fieldsToUpdate = [
+        'street',
+        'postalCode',
+        'city',
+        'latitude',
+        'longitude',
+        'coords',
+        'banId',
+        'search-addressAutocomplete',
+        'addressAutocomplete',
+      ]
+
+      // Empty all fields value
+      await Promise.all(
+        fieldsToUpdate.map((fieldName) => formik.setFieldValue(fieldName, ''))
+      )
+
+      // Make all fields untouched
+      // (This will prevent validation errors to be shown if user previously touched those fields, then switched that trigger OFF, then ON again)
+      await Promise.all(
+        fieldsToUpdate.map((fieldName) =>
+          formik.setFieldTouched(fieldName, false)
+        )
+      )
```

Comment:
Est-ce que le `shouldValidate` n'enlève pas le besoin de faire le setFieldtouched ?

```suggestion
      // Empty all fields value
      await Promise.all(
        fieldsToUpdate.map((fieldName) => formik.setFieldValue(fieldName, '', true))
      )
```

Reply:
Je viens d'essayer, mais non :( 

https://github.com/user-attachments/assets/a3520a02-fabd-4c84-ac58-1d4bbb1f95c6